src/control/server/harness.go

```diff
@@ -186,22 +186,10 @@ func (h *EngineHarness) CallDrpc(ctx context.Context, method drpc.Method, body p
 	// the first one that is available to service the request.
 	// If the request fails, that error will be returned.
 	for _, i := range h.Instances() {
-		if !i.IsReady() {
-			if i.IsStarted() {
-				if err == nil {
-					err = errInstanceNotReady
-				}
-			} else {
-				if err == nil {
-					err = FaultDataPlaneNotStarted
-				}
-			}
-			continue
-		}
 		resp, err = i.CallDrpc(ctx, method, body)
 
 		switch errors.Cause(err) {
-		case errDRPCNotReady, FaultDataPlaneNotStarted:
+		case errEngineNotReady, errDRPCNotReady, FaultDataPlaneNotStarted:
```

Comment:
Could it be useful to add some notice message on the error.
From what I understand, if the `CallDrpc()` function fail with more than one engine, then only the last error will be returned to the caller.

Moreover, I am not sure if the defer function will properly manage such use case.

Reply:
Yes some logging may be beneficial. Whilst in most cases I don't think the defer operating on the last error would cause a problem there may be an edge case where the dRPC failure handlers are not called when they should do so I will create a ticket: DAOS-14214. @mjmac do you think this is worth fixing?

Reply:
If this PR is introducing a window where the dRPC failure handlers may not be invoked correctly, then that needs to be fixed before landing. The failure handlers are critical to correct operation of the MS. In particular, if a server is the current MS leader and there are no engines available to service dRPC requests, then the server must relinquish leadership so that another peer can step up to service MS leader requests.

Reply:
@mjmac what has been captured as DAOS-14214 is not something that has been introduced by this PR, it is a potential issue that has existed for a while.

Reply:
please see updates on DAOS-14214 regarding this issue